packages/trader/src/Stores/Modules/Trading/trade-store.js

```diff
@@ -663,15 +677,16 @@ export default class TradeStore extends BaseStore {
             this.proposal_requests = {};
         }
         if (is_changed_by_user && /\bcurrency\b/.test(Object.keys(obj_new_values))) {
-            const prev_currency =
-                obj_old_values && !isEmptyObject(obj_old_values) && obj_old_values.currency
-                    ? obj_old_values.currency
-                    : this.currency;
-            if (isCryptocurrency(obj_new_values.currency) !== isCryptocurrency(prev_currency)) {
-                obj_new_values.amount =
-                    is_changed_by_user && obj_new_values.amount
-                        ? obj_new_values.amount
-                        : getMinPayout(obj_new_values.currency);
+            const prev_currency = obj_old_values?.currency || this.currency;
+            const has_currency_changed = obj_new_values.currency !== prev_currency;
+
+            const should_reset_stake =
+                isCryptocurrency(obj_new_values.currency) ||
```

Comment:
What about fiat to fiat? Don't we need to reset the stake?

Reply:
stake won't be invalid when switch from fiat to fiat.